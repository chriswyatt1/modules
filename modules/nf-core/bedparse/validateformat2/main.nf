process BEDPARSE_VALIDATEFORMAT2 {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bedparse:0.2.3--py_0':
        'biocontainers/bedparse:0.2.3--py_0' }"

    input:
    tuple val(meta), path(bed)

    output:
    tuple val(meta), path("*.out"), emit: validatedbed
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    bedparse validateFormat \\
        $args \\
        $bed \\
        > ${bed}.out

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bedparse: 0.2.3
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """

    touch ${bed}.out

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bedparse: \$(bedparse --version)
    END_VERSIONS
    """
}
