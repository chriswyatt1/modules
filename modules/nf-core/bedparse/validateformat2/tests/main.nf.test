// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test bedparse/validateformat2
nextflow_process {

    name "Test Process BEDPARSE_VALIDATEFORMAT2"
    script "../main.nf"
    process "BEDPARSE_VALIDATEFORMAT2"

    tag "modules"
    tag "modules_nfcore"
    tag "bedparse"
    tag "bedparse/validateformat2"

    test("sarscov2 - bed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bed - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
