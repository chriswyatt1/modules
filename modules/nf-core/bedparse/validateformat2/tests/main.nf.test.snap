{
    "sarscov2 - bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.out:md5,fe4053cf4de3aebbdfc3be2efb125a74"
                    ]
                ],
                "1": [
                    "versions.yml:md5,89ff3cf90558c7b1dfe1f09327de9b6f"
                ],
                "validatedbed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.out:md5,fe4053cf4de3aebbdfc3be2efb125a74"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,89ff3cf90558c7b1dfe1f09327de9b6f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T11:14:24.721637737"
    },
    "sarscov2 - bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,dd5122848afaa3cb905d6f88497c551c"
                ],
                "validatedbed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dd5122848afaa3cb905d6f88497c551c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T11:09:20.160410573"
    }
}